### PR TITLE
[GenericPoseGraph] Add InterpolatedRelativePose2D constraint.

### PR DESCRIPTION
**Pull request by pifon2a**

### CODE REVIEW

File: cartographer/pose_graph/constraint/interpolated_relative_pose_constraint_2d.cc [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+#include "cartographer/pose_graph/constraint/interpolated_relative_pose_constraint_2d.h"
+
+#include "cartographer/common/make_unique.h"
+#include "cartographer/common/utils.h"
+
+namespace cartographer {
+namespace pose_graph {
+namespace {
+
+void AddPoseParameters(Pose3D* pose, ceres::Problem* problem) {
```
Comment (mbokeloh) [inline cartographer/pose_graph/constraint/interpolated_relative_pose_constraint_2d.cc]: Nit: AddPose3DParameters? also 2d below.

File: cartographer/pose_graph/constraint/interpolated_relative_pose_constraint_2d.h [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+ protected:
+  proto::CostFunction ToCostFunctionProto() const final;
+
+ private:
+  // clang-format off
+  using AutoDiffFunction = ceres::AutoDiffCostFunction<
+      InterpolatedRelativePoseCost2D,
+      6 /* residuals */,
+      3 /* pose variables of first start pose */,
```
Comment (mbokeloh) [inline cartographer/pose_graph/constraint/interpolated_relative_pose_constraint_2d.h]: Nit: '2d pose variables ...'

**Comment by pifon2a** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user pifon2a. Merge queue now has a length of 1.

**Pull request merged**